python/pyspark/pandas/frame.py

```diff
@@ -7637,102 +7706,35 @@ def merge(
         As described in #263, joining string columns currently returns None for missing values
             instead of NaN.
         """
-
-        def to_list(os: Optional[Union[Name, List[Name]]]) -> List[Label]:
-            if os is None:
-                return []
-            elif is_name_like_tuple(os):
-                return [cast(Label, os)]
-            elif is_name_like_value(os):
-                return [(os,)]
-            else:
-                return [o if is_name_like_tuple(o) else (o,) for o in os]
-
         if isinstance(right, ps.Series):
             right = right.to_frame()
 
-        if on:
```

Comment:
The logic from 7654-7693 has been moved verbatim to `self._resolve_merge_key_names`